x/protocolpool/keeper/keeper.go

```diff
@@ -0,0 +1,226 @@
+	}
+	keeper.Schema = schema
+
+	return keeper
+}
+
+// GetAuthority returns the x/protocolpool module's authority.
+func (k Keeper) GetAuthority() string {
+	return k.authority
+}
+
+// GetCommunityPoolModule gets the module name that funds should be sent to for the community pool.
+// This is the address that x/distribution will send funds to for external management.
+func (k Keeper) GetCommunityPoolModule() string {
+	return types.ProtocolPoolEscrowAccount
+}
+
+// FundCommunityPool allows an account to directly fund the community fund pool.
+func (k Keeper) FundCommunityPool(ctx sdk.Context, amount sdk.Coins, sender sdk.AccAddress) error {
+	return k.bankKeeper.SendCoinsFromAccountToModule(ctx, sender, types.ModuleName, amount)
+}
+
+// DistributeFromCommunityPool distributes funds from the protocolpool module account to
+// a receiver address.
+func (k Keeper) DistributeFromCommunityPool(ctx sdk.Context, amount sdk.Coins, receiveAddr sdk.AccAddress) error {
+	return k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, receiveAddr, amount)
+}
+
+// GetCommunityPool gets the community pool balance.
+func (k Keeper) GetCommunityPool(ctx sdk.Context) (sdk.Coins, error) {
+	moduleAccount := k.authKeeper.GetModuleAccount(ctx, types.ModuleName)
+	if moduleAccount == nil {
+		return nil, errorsmod.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", types.ModuleName)
+	}
+	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
+}
+
+// DistributeFunds sets the amount to be distributed among recipients.
+// Get all valid continuous funds:
+// - for each continuous fund, check if expired and remove if so
+// - for each continuous fund, distribute funds according to percentage
+// - distribute remaining funds to the community pool
+//
+// This function is run at the BeginBlocker method and therefore must be very safe.
+func (k Keeper) DistributeFunds(ctx sdk.Context) error {
+	// Get current balance of the intermediary module account
+	moduleAccount := k.authKeeper.GetModuleAccount(ctx, types.ProtocolPoolEscrowAccount)
+	if moduleAccount == nil {
+		return errorsmod.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", types.ProtocolPoolEscrowAccount)
+	}
+	params, err := k.Params.Get(ctx)
+	if err != nil {
+		return err
+	}
+
+	// only take into account the balances of denoms whitelisted in EnabledDistributionDenoms
+	amountToDistribute := sdk.NewCoins()
+	for _, denom := range params.EnabledDistributionDenoms {
+		bal := k.bankKeeper.GetBalance(ctx, moduleAccount.GetAddress(), denom)
+		amountToDistribute = append(amountToDistribute, bal)
+	}
+
+	// if the balance is zero, return early
+	if amountToDistribute.IsZero() {
+		return nil
+	}
+
+	remainingCoins := sdk.NewCoins(amountToDistribute...)
+	iter, err := k.ContinuousFunds.Iterate(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create iterator for continuous funds: %w", err)
+	}
+
+	kvalues, err := iter.KeyValues()
```

Comment:
we can just do `iter.Values()`

Reply:
We actually cannot do this because the reciepient value on the Fund value is a `string` and we want to use the key (`sdk.AccAddress`)